### ISSUE TITLE
Parser trips up on common snowflake query history

### ISSUE THREAD

**Issue by peteclark3** [tokern/data-lineage#78]
Currently, the parser trips up on many common snowflake query history entries like `select query_text from table(information_schema.query_history());` - also queries with the `rm @SNOWFLAKE_...` syntax... also queries with the keyword `recluster` ... in the latter case, the error being `syntax error at or near "recluster", at index 35` ... I am systematically removing these from analysis prior to sending to the analyzer but just FYI that without doing this, the analyzer throws an exception

**Comment by vrajat** [comment]
Yes. This is a known issue and happy to discuss how to approach parsing Snowflake queries better. Right now, this package uses [pglast](https://github.com/lelit/pglast) for parsing. So it parses Postgres syntax. It also does pretty well with Redshift since the dialect is very similar. 
A set of parsers are required to support different databases. There are a few options that I have considered and funded through freelance gigs:

* Use Apache Calcite and add support for custom syntax. This worked pretty well for a gig but it is in Java. Since this package is in Python, Python/Java bridges are a pain to use.
* Use [lark](https://github.com/lark-parser/lark) and build a parser. Note that one doesnt have to start from scratch. 

Both of these require substantial effort and I am looking for opportunities to fund it.
